src/documents/PromisesAPI.md

```diff
@@ -0,0 +1,236 @@
+
+* `listener {Function:(previousValue {Any}, options {Object}) →  Any|Promise}` where `previousValue` is the resolved return from the previous listener notified in the chain,
+or the initial `payload` value supplied to `fluid.promise.fireTransformEvent` if it is the first in the chain, and `options` is the last argument to `fluid.promise.fireTransformEvent`.
+
+## Inverse API recognised by promises consumed by sequential algorithms
+
+Both `fluid.promise.sequence` and `fluid.promise.fireTransformEvent` will recognise the following method supplied by the user on any promise
+returned by one of sources in the sequence:
+
+### promise.accumulateRejectionReason(error)
+
+* `error: {Object|Error}` A rejection which has been received from a promise "to the right` of this one in a promise sequence.
+* Returns: `{Object|Error}` A rejection reason which has been "wrapped" or "decorated" in some way in order to add information about the
+function of ***this*** promise. For example, if this promise was intended to resolve by reading a file from the filesystem, the rejection reason
+could be decorated with a string like "while reading file Xxxxx". It's important that the user's implementation preserves all the information in 
+the original rejection reason - if it contains a string `message`, it should be prefixed or suffixed with the additional information, or if it 
+contains an error stack, it should be left untouched.
+
+This is a form of "inverse API". The promise API does not implement this method, but it can be implemented by any consumer of 
+promises by adding a function with this signature named `accumulateRejectionReason` to a promise object. This method is only
+relevant when consuming a sequence of promises using one of Infusion's sequential [promise algorithms](#promise-algorithms).
+
+Let us imagine the promises in a sequence (array) laid out from left to right, in order of sequential execution. 
+This method is called by a sequential promise algorithm when a promise somewhere in the sequence has rejected. Ordinarily, execution would
+pass directly to the overall rejection handler for the sequence. However, before this happens, the sequence algorithm will pass from
+*** right to left *** from the point of rejection and inspect each of the promises in that section for an `accumulateRejectionReason` implementation.
+
+If an implementation is found, it will be called with the current rejection reason as an argument, and the return value will be used as the new
+rejection reason. The resolution algorithm then continues to the left with this new rejection reason in place, etc. Finally the fully accumulated
+rejection reason will be dispatched to the overall rejection handler.
+
+What familiar exception-handling pattern from synchronous code does this reproduce? It is the *** rethrowing pattern ***, described in the Java context by 
```

Comment:
This commentary would be better moved out of the API documentation, perhaps into the wiki somewhere?